bpfprogs/probes.go

```diff
@@ -86,7 +78,54 @@ func GetProgramSectionDetails(sectionName string) (string, string, string) {
 		return sections[0], sections[1], sections[2]
 	case models.KProbe, models.KRetProbe:
 		return sections[0], sections[1], ""
+	case models.UProbe, models.URetProbe:
+		var funcName string
+		if len(sections) > 2 {
+			funcName = strings.Join(sections[1:], "/")
+		}
+		return sections[0], funcName, ""
 	default:
 		return "", "", ""
 	}
 }
+
+func (b *BPF) AttachProbePerfEvent(hookName, progType string, prog *ebpf.Program) (link.Link, error) {
+	var kp link.Link
+	var err error
+	switch strings.ToLower(progType) {
+	case models.KProbe:
+		kp, err = link.Kprobe(hookName, prog, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to link kprobe hook name %s error %v", hookName, err)
+		}
+	case models.KRetProbe:
+		kp, err = link.Kretprobe(hookName, prog, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to link kretprobe hook name %s error %v", hookName, err)
+		}
+	case models.UProbe, models.URetProbe:
+		funcNames := strings.Split(hookName, ":")
+		var symbol string
+		ex, err := link.OpenExecutable(funcNames[0])
+		if err != nil {
+			return nil, fmt.Errorf("failed to openExecutable binary file %s error %v", hookName, err)
+		}
+
+		if len(funcNames) == 1 {
+			symbol = funcNames[0]
+		} else {
+			symbol = funcNames[len(funcNames)-1]
+		}
```

Comment:
Can we make the above code as function, like symbol = getSymbol(hookName)?  Therefore eliminate the if else below and keep the case statement for Uprobe and URetProbe separated.